### ISSUE TITLE
Improvements of the Spyder logo

### ISSUE THREAD

**Issue by jnsebgosselin** [spyder-ide/spyder#6509]
This issue is related to the work initiated in PR #6352. The goal is to document modification done to the logo in order to improve it visually and to make it more friendly for light background. Below are a list of ideas on which I would like us to work on in order to improve the logo.

- Improve the snake
- Make the web symetrical
- Make the logo looks better on the light background
- Try to enclose the logo in a circle.
- Make the current red snake into an S (similar to the Scipy symbol)

**Comment by jnsebgosselin** [comment]
As a starter, I've done some work to make the web symmetrical.

![spyder_symetrical_web](https://user-images.githubusercontent.com/10170372/36637576-d41d2f5c-19ac-11e8-86ed-61782bd44c0b.png)

**Comment by jnsebgosselin** [comment]
Some attempt at enclosing the logo...

![spyder_enclosed](https://user-images.githubusercontent.com/10170372/36638101-53b4469a-19b9-11e8-8335-154714edc538.png)

And below is how it looks in the banner with light or dark background.

![spyder_enclosed_banner](https://user-images.githubusercontent.com/10170372/36638120-f507ac80-19b9-11e8-8312-5c97dd88d428.png)

**Comment by bcolsen** [comment]
I really like how the webs go out to the edges. It gives great contrast to the webs

I thought I would give it a shot.  I wanted to preserve the icon shape and curve the webs to make them flow. The snake isn't working for me. Maybe I'm to used to the simple snake :-)

![spyder_curve](https://user-images.githubusercontent.com/10513354/36639657-b8f4aa1c-19ce-11e8-9cf1-7ae161151fc6.png)

**Comment by jnsebgosselin** [comment]
@bcolsen Nice work! I like the web. Could you make an alternate version with the same dark grey everywhere and with the original snake please? Also can you try to use full white for the web instead.

I've also try with various type of snake and everyone looked strange... Also using the snake to make an "S" never gave good result for me because it felt like too easy. I think we need to make a snake that is similar the the actual one, but to improve it a little, like for the shape of the head for example.

Would you like to give it a shot?

![spyder_enclosed_snake](https://user-images.githubusercontent.com/10170372/36642491-f20bbb8a-1a0d-11e8-8f85-daf540f543f2.png)

**Comment by jnsebgosselin** [comment]
@bcolsen Could you also show how it looks on a dark background please. That would be nice thank you.

**Comment by ccordoba12** [comment]
@jnsebgosselin, I like a lot the squared shape you gave to the icon. What if you combine it with the S from @bcolsen and give the snake's head the shape of your third iteration from left to right (the one with several curls)?

**Comment by jnsebgosselin** [comment]
@ccordoba12 OK, I'll try to combine both concepts, it is a good idea. I would like to try @bcolsen 's web in a squared shape like I did. I think that would look great. I'll also iterate some more with the S shaped snake idea.

@bcolsen would you agree to share with me your design in a svg format so that I can more easily play with it? That would be awesome. You can download what I have done so far here:

https://www.dropbox.com/sh/m78l0qf1w070use/AACyD9hV-v9yl4Zi0Firhh7ra?dl=0

**Comment by jnsebgosselin** [comment]
![spyder_snake_sshaped](https://user-images.githubusercontent.com/10170372/36644317-4d15a9f6-1a26-11e8-9147-b3690dc1917e.png)

**Comment by ccordoba12** [comment]
Getting closer, getting closer! This is really cool!

What I still like from @bcolsen design is that you can clearly distinguish the S *and* the red snake. In your design, I think only the snake is distinguishable.

Could you make the snake's head to be at the end of the S's top arm (I don't know how describe it better :-)? I mean, instead of going perpendicular to it. And the same for the snake's tail? Thanks!

I also think the S can occupy more space inside the square, like in @bcolsen design.

**Comment by ccordoba12** [comment]
My idea is to make a contrast with RStudio's logo (between the S and R, I mean).

**Comment by jnsebgosselin** [comment]
@ccordoba12 I understand, good idea. I'm going to try it.

**Comment by bcolsen** [comment]
Here's a link:

https://drive.google.com/file/d/1GO7F7D3SNfdo_An8ngMryPLE8PPMVLxM/view?usp=sharing

I used the lighter background in the web to increase the contrast of the snake.  Something like that and a darker snake may let us avoid the outline on the snake

**Comment by homocomputeris** [comment]
> Make the web symetrical

IMO, it's nice as it is. Symmetry is less, well, original.
> Make the current red snake into an S (similar to the Scipy symbol)

Python's snake logo can be used to make an S too.

**Comment by jnsebgosselin** [comment]
Thank you very much @bcolsen! Below are some more concepts integrating @ccordoba12 suggestions and @bcolsen web design.

![spyder_snake_sshaped2](https://user-images.githubusercontent.com/10170372/36647076-17cbae80-1a4e-11e8-8a87-118283457478.png)

**Comment by jnsebgosselin** [comment]
> IMO, it's nice as it is. Symmetry is less, well, original.

My designs are going to be symmetrical... sorry but I can't help it. It's not like its a game changer anyway...

Following @homocomputeris suggestion, here is a quick attempt to make a design that recall the snake in the Python's logo. Some work is still needed to make the S looks more like a S.

![spyder_snake_sshaped_pythonlogo](https://user-images.githubusercontent.com/10170372/36647488-66804120-1a54-11e8-9dc2-5229993a31cc.png)

**Comment by jnsebgosselin** [comment]
Some modified version of the designs above.

![spyder_snake_sshaped_pythonlogo2](https://user-images.githubusercontent.com/10170372/36650106-892bf884-1a6f-11e8-890c-cb38e3adafcc.png)

**Comment by jnsebgosselin** [comment]
![spyder_snake_sshaped_pythonlogo3](https://user-images.githubusercontent.com/10170372/36651912-66e4c184-1a79-11e8-8ed9-44dbdaa0e823.png)

**Comment by bcolsen** [comment]
Wow I really like 10.   The snake is great upside down and the over line is I liked the snake going through web in 6 as well.

I like the idea of the integration of the logo with the name too, and the shortened subtitle as well.  The S needs more contrast though, because on first glance I just see the "pyder"

A pallet swap gives an "S" first web later thing:

![pallet swap](https://user-images.githubusercontent.com/10513354/36657545-49de806e-1a8a-11e8-89bf-a6b5d1dbc879.png)

Flat font to match the "S" is [Cabin Bold](https://fonts.google.com/specimen/Cabin?selection.family=Cabin) under  the Open Font License.

Without Subtitle:

![pallet swap sans](https://user-images.githubusercontent.com/10513354/36657906-a8406ba8-1a8b-11e8-9979-2baf21f5e94e.png)

That would look sweet on a skate board or even an IDE! (I guess)

**Comment by jitseniesen** [comment]
This is really starting to look cool. Keep it up!

**Comment by jnsebgosselin** [comment]
Nice observations @bcolsen! I want a sticker to but on the back of my Alienware lol

Here is a small twist to what you've done if we want to keep a white background so that it is easier to integrate as a banner on the website or on the readme.

![spyder - snake - sshaped - banner](https://user-images.githubusercontent.com/10170372/36659331-2432e900-1aa2-11e8-84c7-86dde00c9c23.png)

**Comment by bcolsen** [comment]
I really do like the white web the best.

To increase the "S" contrast I tried to in crease the size a bit. Small one on the left bigger on right:

![rect2887-6-7-9-2](https://user-images.githubusercontent.com/10513354/36704675-90cd0466-1b1e-11e8-829f-73d9c0d88ccd.png)


The "s" a bit more visible when it's bigger.  

I don't know what font you're using now, but I think I like the "R" in cabin a bit more

**Comment by goanpeca** [comment]
I like the general ideas but, the red is a really bad color, maybe we can change it :-) ?

**Comment by bcolsen** [comment]
There are a lot more green snakes than red snakes :-)

People can see green better too.

**Comment by goanpeca** [comment]
Yeah, we had thought about using green as well, could we see some options :-) ?

**Comment by jnsebgosselin** [comment]
@goanpeca What about yellow? Yellow would look great also no?

![image](https://user-images.githubusercontent.com/10170372/36706176-af55d300-1b36-11e8-9ae6-b0d86c7ae57e.png)

**Comment by goanpeca** [comment]
@jnsebgosselin sure, lets just get away from the red once and for all please, its like the worst color ever!

**Comment by bcolsen** [comment]
ooh yeah...burnt yellow and grey..my favorite color combo

 A (I think this is the most clear title..easy to see spyder right away):
![pallet swap yelow](https://user-images.githubusercontent.com/10513354/36707727-cffef8e8-1b2c-11e8-86d2-b8918726442c.png)

with yellow text:

B:
![pallet swap yelow text](https://user-images.githubusercontent.com/10513354/36707816-34af3546-1b2d-11e8-8161-ac7b6b7875c7.png)

Skateboard:

![pallet swap skate](https://user-images.githubusercontent.com/10513354/36707955-e4faa4b2-1b2d-11e8-9fd4-526384eb7400.png)

**Comment by bcolsen** [comment]
As an 48px icon:

![pallet swap icon](https://user-images.githubusercontent.com/10513354/36709259-2d9a2778-1b34-11e8-81cd-c9d1c8901845.png)

**Comment by jnsebgosselin** [comment]
I really really like the colors, but it doesn't feel like Spyder to me. I'm not against it, but I'm not sure it is a good idea to use another color than red for the snake. The red snake kind of grow on me over the years using Spyder... I'll look at it again in a couple of days. Maybe I'll have another perspective.

The logo alone with the red snake doesn't look so bad. Maybe we can rework the banner so that there is less red?

**Comment by jnsebgosselin** [comment]
Gotta love the colors :D Thanks everyone for the great collaboration/interaction we're having on this one. This is fun.

![spyder_banner_colors](https://user-images.githubusercontent.com/10170372/36752488-3a30ef38-1bd1-11e8-91e0-a54647c3616e.png)

**Comment by bcolsen** [comment]
> Gotta love the colors :D

Yeah I think things are working out well!

The red snake has strong branding. Here's one with a darker red. It's a little easier on the eyes than the original red(seen on the left icon).

![rect2887-6-7-9-2dark](https://user-images.githubusercontent.com/10513354/36747332-03a3da94-1bb2-11e8-95ac-c186913d7e69.png)

could go more orange:

![rect2887-6-7-9-2_orange](https://user-images.githubusercontent.com/10513354/36748082-09b4fb50-1bb4-11e8-96dc-4180a0e82001.png)

something a little different:

![rect2887-6-7-9-2_shift](https://user-images.githubusercontent.com/10513354/36752990-02d1849e-1bc2-11e8-880e-a48a3592450a.png)

**Comment by bcolsen** [comment]
Here is a link to the big S

https://drive.google.com/open?id=1szVQf1Pd5rjTlrySc_bAdeC9OWdsDhs9

**Comment by jnsebgosselin** [comment]
> The red snake has strong branding. Here's one with a darker red. It's a little easier on the eyes than the original red(seen on the left icon).

I agree with you. That is why I'm hesitant to completely drop the red...

I like the red design you've made. It looks good and you are right it is a lot easier on the eye. Following your idea, here is another 3-colors design I've made with a different red. I've also enlarged the S like you suggested.

![spyder_banner_red_alt](https://user-images.githubusercontent.com/10170372/36764237-ad81df56-1bf9-11e8-979c-8e17f8530e7a.png)

**Comment by bcolsen** [comment]
The tan in the middle one gives good contrast with the red.

I tried it with a solid red back with a dark snake for fun, but it was too much red, but just in the middle works:

![pallet swap red back](https://user-images.githubusercontent.com/10513354/36767781-33200512-1bf9-11e8-948a-630b105f5613.png)

Also, are we confident enough in our IDE for this one:

![pallet swap pink](https://user-images.githubusercontent.com/10513354/36768051-61aa0f58-1bfa-11e8-9efa-0eba26e8b0cd.png)

That's light red BTW ;-)

**Comment by jnsebgosselin** [comment]
I showed our concepts to some of my colleges, one of which is very good with graphic art. Overalls, the feedback was not as good as I would have wished.... Basically they said it was "interesting..." (not good hehe), but that the contrast with the S and the web was generally bad, that the "P" overlapping the logo is generally a bad thing, and that the overall design did not felt very modern... and well, in retrospect, I kind of agree with them ...

So I went back to the drawing board, trying out something completely different. I used the snake from the Python logo and tried something different for the web.

![image](https://user-images.githubusercontent.com/10170372/36787283-9af3d930-1c57-11e8-95e9-77a9f57dd6c0.png)

![image](https://user-images.githubusercontent.com/10170372/36787477-459bf21e-1c58-11e8-9b66-e53be0be9cd3.png)

![image](https://user-images.githubusercontent.com/10170372/36787328-c904179a-1c57-11e8-87e2-11c5194a61ce.png)

**Comment by jnsebgosselin** [comment]
Same idea, better web.

![image](https://user-images.githubusercontent.com/10170372/36806301-9f1ab2d2-1c8d-11e8-9af8-c647482256cd.png)

**Comment by goanpeca** [comment]
Thanks guys for working on this :-) 

I think we should move away from the Python wording, since the idea is to support more languages, and saying it is written in python is also a secondary thing....  so we should start embracing

`Spyder - The Scientific IDE` or something else...

**Comment by jnsebgosselin** [comment]
@goanpeca  ok... well this goes for the snake in the logo too. There is not really any reason to keep a snake in there if we want to move away from the fact that Spyder is an IDE mainly for Python. Also, the name Spyder, as an acronym, becomes a little bit less interesting, so is a change in the name also conceivable?

Until these questions are clarified, I think it is a waste of time to work on a new logo and banner.

**Comment by ccordoba12** [comment]
> I think we should move away from the Python wording, since the idea is to support more languages, and saying it is written in python is also a secondary thing

Given our very limited resources, I really think we should only focus in Python. Only Jupyter has enough resources to support more than Python. And out of Julia, R and C++, the support is very limited.

**Comment by goanpeca** [comment]
>ok... well this goes for the snake in the logo too.

I think the snake is fine... cause it is built with python and the web of plugins and modules still applies, I meant specifically `The Python Scientific IDE` wording.

**Comment by goanpeca** [comment]
Also @jnsebgosselin @bcolsen I really appreciate all the work you are putting on this, and I agree this is fun, but the truth is we are not graphic designers and anything we come up will probably have issues or be suboptimal, but the idea we are aiming at is pretty clear. We really need to take it from here to a professional designer.

If you know someone I am all ears to arrange the costs.

**Comment by jnsebgosselin** [comment]
@goanpeca yep I agree with you. We've laid some interesting ideas so we can take it to another level. I don't know any professional designer unfortunately. If you don't mind, I'll let you handle this one from then so that I can focus on other things.

**Comment by jnsebgosselin** [comment]
@goanpeca 
This guy http://sebastienthibault.com/ is very good. He is from my hometown.

also https://www.fiverr.com/

**Comment by juanis2112** [comment]
@isabela-pf this is the thread for the discussion of the actual logo of Spyder so you can take a look at some ideas we had at first.
